src/VisualStudio/CSharp/Impl/ProjectSystemShim/TempPECompilerService.cs

```diff
@@ -59,7 +59,9 @@ public void CompileTempPE(string pszOutputFileName, int sourceCount, string[] fi
                     .WithXmlReferenceResolver(XmlFileResolver.Default)
                     .WithMetadataReferenceResolver(metadataResolver));
 
-            compilation.Emit(pszOutputFileName);
+            var result = compilation.Emit(pszOutputFileName);
+
+            Contract.ThrowIfFalse(result.Success);
```

Comment:
Looking at what the old TempPE code did, it would return an HRESULT to the project system with the failure. It wouldn't do anything else like logging messages or anything.